### ISSUE TITLE
[windows] Can leave the machine without a puppet installed

### ISSUE THREAD

**Issue by liamjbennett** [opentable/puppet-puppetversion#3]
If there is a issue with the network in downloading the new version of puppet then the script will continue and uninstall puppet leaving the box in a state where it has no installation of puppet at all.

**Comment by ferventcoder** [comment]
@liamjbennett any plans to harden this?

**Comment by liamjbennett** [comment]
@ferventcoder yes we are looking into it. I am thinking about moving this module the community, but I think I need to discuss it on the mailing list first.
